sentry_dsn_data.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/getsentry/sentry-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DSN = "dsn"
+
+// map from Sentry DSN to Client
+type DsnData struct {
+	mutex     sync.RWMutex
+	clientMap map[string]*sentry.Client
+}
+
+func NewDsnData() *DsnData {
+	return &DsnData{
+		mutex:     sync.RWMutex{},
+		clientMap: make(map[string]*sentry.Client),
+	}
+}
+
+// return client if added successfully
+func (d *DsnData) AddClient(options sentry.ClientOptions) (*sentry.Client, error) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	// check if we already encountered this dsn
+	existingClient, ok := d.clientMap[options.Dsn]
+	if ok {
+		return existingClient, errors.New("a client with the given dsn already exists")
+	}
+
+	// create a new client for the dsn
+	newClient, err := sentry.NewClient(
+		sentry.ClientOptions{
+			Dsn:              options.Dsn,
+			Debug:            true,
+			AttachStacktrace: true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	d.clientMap[options.Dsn] = newClient
+
+	return newClient, nil
+}
+
+// retrieve a client with given dsn
+func (d *DsnData) GetClient(dsn string) (*sentry.Client, error) {
```

Comment:
Given the implementation and map semantics, we can use `bool`, not `error` here.